libs/common/src/main/java/org/opensearch/common/round/BinarySearcher.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.round;
+
+import org.opensearch.common.annotation.InternalApi;
+
+import java.util.Arrays;
+
+/**
+ * It uses binary search on a sorted array of pre-computed round-down points.
+ *
+ * @opensearch.internal
+ */
+@InternalApi
+class BinarySearcher implements Roundable {
+    private final long[] values;
+    private final int size;
+
+    BinarySearcher(long[] values, int size) {
+        assert size > 0 : "at least one value must be present";
```

Comment:
Same as above...I'd just unconditionally throw an IllegalArgumentException instead of using assertions.